project/frontend/src/App.js

```diff
@@ -37,11 +40,62 @@ function App() {
     setUser((prevState) => {
       return {
         ...prevState,
-        id: 123,
+        id: 1,
       };
     });
   }
 
+  React.useEffect(() => {
+    if (user.signedIn && user.id && compId) {
+      Promise.all([
+        setLoading(true),
+        fetch("./competitionInfo?user=" + user.id + "&competition=" + compId)
+          .then((response) => response.json())
+          .then((data) => {
+            setCompetitionInfo((prevState) => {
+              return {
+                ...prevState,
+                generalInfo: data,
+              };
+            });
+          }),
+        fetch("./recentBuys?competition=" + compId)
+          .then((response) => response.json())
+          .then((data) => {
+            setCompetitionInfo((prevState) => {
+              return {
+                ...prevState,
+                recentBuys: data,
+              };
+            });
+          }),
+        fetch("./investments?user=" + user.id + "&competition=" + compId)
+          .then((response) => response.json())
+          .then((data) => {
+            setCompetitionInfo((prevState) => {
+              return {
+                ...prevState,
+                investments: data,
+              };
+            });
+          }),
+        fetch("./trending")
+          .then((response) => response.json())
+          .then((data) => {
+            setCompetitionInfo((prevState) => {
+              return {
+                ...prevState,
+                trending: data,
+              };
+            });
+          }),
```

Comment:
I feel like the logic in these fetch calls could be extracted into a function that returns a promise, for example
`fetchAndUpdateCompetitionInfo(apiEndpoint, stateKey)`

And also using ES6 [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)

Would end up looking something like
```js
fetchAndUpdateCompetitionInfo(
    `./competitionInfo?user=${user.id}&competition=${compId}`, 
    "generalInfo")
```

Reply:
Just to clarify, which promise do you mean it should return at? The response promise or the jsonified promise?

Reply:
as in something like
```js
Promise.all([
  setLoading(true),
  fetchAndUpdateCompetitionInfo(`./competitionInfo?user=${user.id}&competition=${compId}, "generalInfo"),
  fetchAndUpdateCompetitionInfo("./trending", "trending"),
  // ... and so on
]).then(() => {
```

Reply:
Done.

Reply:
Please note that this was actually part of PR #95 (which I haven't merged) and this PR was only the last 4 commits before your comment (as indicated in the PR description) so the changes you requested were made in the branch of PR #95, and subsequently pulled from that branch so you can see them here as well - hence the merges